src/Data/UTxO/Transaction.hs

```diff
@@ -125,6 +125,8 @@ data ErrMkPayment
         -- ^ Payments must have at least one output
     | MissingSignature
         -- ^ Payments must have a signature for each input.
+    | TooLowOutput
+        -- ^ Each payment must not be smaller than minUTxOvalue
```

Comment:
Perhaps `TooSmallOutput` ?

Reply:
Although, how would we know that from cardano-transactions :thinking:  ? This depends on protocol parameters which this library is not aware of.

Reply:
if we do not set this here, then if someone constructs output which is  < minUTxOvalue and use external tx endpoint then node will reject it and 500 will be emitted (at this moment). So I decided to introduce it here. But I agree it is not so straightfoward

Reply:
> then node will reject it and 500 will be emitted (at this moment)

Indeed and that is totally fine. We could make the error reporting on cardano-wallet better, but cardano-transactions isn't the place to fix this :+1:

Reply:
agree, handling/exposing node errors in for example external tx endpoint is sound approach. reverted this change in https://github.com/input-output-hk/cardano-transactions/pull/17/commits/51ce57e3a1225b95303143a429a4a4d98b9a501b